File: CreationalPatterns/FactoryMethod/TypeScript/NotificationCreator.ts
import { Notifier } from './Notifier';

export abstract class NotificationCreator {
  // Factory Method
  abstract factoryMethod(): Notifier;

  // Core logic that uses the Notifier
  send(message: string): void {
    const notifier = this.factoryMethod();
    notifier.send(message);
  }
}

File: CreationalPatterns/FactoryMethod/CSharp/NotificationCreator.cs
namespace FactoryMethod {
    public abstract class NotificationCreator {
        // Factory Method
        public abstract INotifier FactoryMethod();

        // Core logic that uses the Notifier
        public void Send(string message) {
            var notifier = FactoryMethod();
            notifier.Send(message);
        }
    }
}

